### PR TITLE
Fix bug related to returnDatasetFieldTypes in Dataverses.listMetadataBlocks endpoint

### LINKED ISSUE #10984: returnDatasetFieldTypes=true not working on non-root collection inheriting "citation" from root
To reproduce:

- Create a collection with an alias/identifier of "testCollection"
- Publish it
- Go to the following URL: http://localhost:8080/api/dataverses/testCollection/metadatablocks?onlyDisplayedOnCreate=true&returnDatasetFieldTypes=true

Expected results: "fields" should be populated with title, etc. from the citation block.

Actual results: "fields" is an empty object

```
{
  "status": "OK",
  "data": [
    {
      "id": 1,
      "name": "citation",
      "displayName": "Citation Metadata",
      "displayOnCreate": true,
      "fields": {}
    }
  ]
}
```

See also:

- #10985

You can see this in action at https://beta.dataverse.org/api/dataverses/testCollection/metadatablocks?onlyDisplayedOnCreate=true&returnDatasetFieldTypes=true

If you go to https://beta.dataverse.org/spa/datasets/testCollection/create not how no fields are available to fill in:

Screenshot 2024-10-30 at 5 59 01 PM

### PR DESCRIPTION
**Pull request by pdurbin**
**What this PR does / why we need it**:

We should be checking for inheritance of metadata blocks from parent collection to child collection.

**Which issue(s) this PR closes**:

- Closes #10984

**Special notes for your reviewer**:

@pdurbin started this pull request but @GPortas added the actual fix.

We were attempting to retrieve the metadata block fields for the requested dataverse without verifying whether isMetadataBlockRoot was set to true. See https://github.com/IQSS/dataverse/pull/10985/files#diff-171c256be8744ca0742a9570ed33a3e335eae0544d08be04545ef31fdd5328d1. Consequently, when accessing a dataverse that inherits from the root, we couldn't find a relationship with the desired metadata block, as that relationship with the block exists in the parent, not the child dataverse.

We have fixed this to always get the fields from the parent collection when isMetadataBlockRoot is false.

**Suggestions on how to test this**:

Make sure Jenkins reports all test passing.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

No. In practice, this only affects the SPA, which is still in development.

**Additional documentation**:

None.

### CODE REVIEW

**Comment by pdurbin** [comment]
As expected https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10985/1/testReport/edu.harvard.iq.dataverse.api/DataversesIT/testListMetadataBlocks/ failed with this:

```
1 expectation failed.
JSON path data[0].fields.title.displayOnCreate doesn't match.
Expected: <true>
  Actual: null
```

This is because the "fields" object is empty:

```
{
    "status": "OK",
    "data": [
        {
            "id": 1,
            "name": "citation",
            "displayName": "Citation Metadata",
            "displayOnCreate": true,
            "fields": {
                
            }
        }
    ]
}
```

**Comment by pdurbin** [comment]
There are failing tests at https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10985/5/testReport/ but they are expected. The tests are:

- https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10985/5/testReport/edu.harvard.iq.dataverse.api/HarvestingClientsIT/testHarvestingClientRun_AllowHarvestingMissingCVV_True/
- https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-10985/5/testReport/edu.harvard.iq.dataverse.api/HarvestingClientsIT/testHarvestingClientRun_AllowHarvestingMissingCVV_False/

The fix is coming in this PR:

- #10990

There are failing tests from GitHub Actions from maven_unit_test.yml:

- https://github.com/IQSS/dataverse/actions/runs/11613822263/job/32340688496?pr=10985

The fix is coming in this PR:

- #10987

**Review by pdurbin — COMMENTED** [review]
Because I created this pull request (just adding a test and a FIXME), I can't approve it in GitHub but I would if I could!

**Review by GPortas — APPROVED** [review]

**Comment by ofahimIQSS** [comment]
Testing Completed - no issues found during testing. 
[Testing of 10985.docx](https://github.com/user-attachments/files/17620098/Testing.of.10985.docx)

**Pull request merged**
